src/content/sidebar/sidebar-wrapper.tsx

```diff
@@ -1,59 +1,130 @@
 import * as browser from "webextension-polyfill"
 
-import {css} from "@emotion/react"
-import {slide as Slider} from 'react-burger-menu'
-import {useEffect, useState} from "react"
+import {useEffect, useState, useRef} from "react"
 import {useTabLocalState} from "../../core/react"
 
 export const SidebarWrapper = () => {
+    const ref = useRef()
+
     const [isOpen, setOpen] = useTabLocalState("sidebarOpen", false)
+
+    // the iframe should be rendered as soon as `isOpen` is true but
+    // when `isOpen` becomes `false` the iframe should only dissapear
+    // after the hiding animation is completed 
+    const [showIFrame, setShowIFrame] = useState(isOpen)
+
     // todo show a loading indicator instead of emptiness
     const [url, setUrl] = useState("")
 
+    const [width, setWidth] = useState(15)
+    const [isLeft, setLeft] = useState(false)
+    const [isDragging, setDragging] = useState(false)
+
     useEffect(() => {
         const messageCallback = (event) => {
             console.log('content-script container ev', event)
             if (event.type === 'update-panel-url') {
                 setUrl(event.url)
-            } else if (event.type === "toggle-sidebar") {
+            // unfortunatly the messages send by Window.postMessage() have a slightly 
+            // different structure than the messages send by the browser api
+            //
+            // we should probably make sure that the message was send by the
+            // url in `url` since every website can post a message like this
```

Comment:
why do we need to listen to both?
as I understand it - we add the new listener because we have to redirect events ? (a bit surprised that the browser events don't get to the element in the shadow dom 🤔)
but if so we only get new events and so should only have a listener for those?

Reply:
since the close button is now _in_ the sidebar we need to listen for both

Reply:
but I wrote a separate listener for this case that also verifies the sender